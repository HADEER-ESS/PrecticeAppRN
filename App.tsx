import React from 'react';
import {Alert, View} from 'react-native';
import TextComp from './src/component/TextComp';
import ButtonComp from './src/component/ButtonComp';


function App(): JSX.Element {
 
  return (
    <View >
      <TextComp Title={"Training App"}/>
      <ButtonComp btnText={"Click ME!!"} btnPressFunction={()=>{Alert.alert("BTN Clicked" , "Hello from btn")}}/>
    </View>
  );
}


export default App;
