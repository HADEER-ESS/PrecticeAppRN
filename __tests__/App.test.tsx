/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: import explicitly to use the types shiped with jest.
import {it, describe, expect} from '@jest/globals';

// Note: test renderer must be required after react-native.
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

describe('<App/>', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<App />);
    expect(tree).toMatchSnapshot()
  });
  it('has two child', () => {

    const tree = renderer.create(<App />).toJSON() as ReactTestRendererJSON
    // const getChildren = tree.toJSON() as ReactTestRendererJSON
    // const childrenOfScreen =  getChildren.children
    // const getLength = getChildren.children as ReactTestRendererJSON[]

    // console.log("Children on screen " , getChildren?.children?.length)
    
    // getChildren?.children?.forEach(node => {

    //   console.log("NODE..... " , node)
    // })

    //count number of Views in Screen
    // expect(getChildren?.children?.length).toEqual(2);
    expect(tree?.children?.length).toBe(2)
  });
});
