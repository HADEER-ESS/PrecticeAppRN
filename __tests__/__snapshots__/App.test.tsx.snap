// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> renders correctly 1`] = `
<View>
  <Text>
    Training App
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#ff77ff",
        "borderRadius": 17,
        "marginHorizontal": "21%",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "#ffffff",
          "fontSize": 24,
          "padding": 11,
          "textAlign": "center",
        }
      }
    >
      Click ME!!
    </Text>
  </View>
</View>
`;
